cowprotocol/cowswap
# fix(permit): report wrong permit to sentry for limit and swap flows

# Summary

Follow up to https://github.com/cowprotocol/cowswap/pull/3562

Context: [thread](https://cowservices.slack.com/archives/C036G0J90BU/p1703777097225409?thread_ts=1703764369.248749&cid=C036G0J90BU)

There was a sentry report of this race condition, but only for limit orders flow. Swap flow did not have it.

# To Test

This is not easily observable and happens very rarely.
There are no clear reproduction steps other than placing orders for tokens that are already approved and checking whether they contain any prehook.

Anyway, the expected flow should work as usual:

1. Pick a token that's permittable and there's no approval for
2. Sell it
* Should request for allowance signature
* Should place order
* Should execute permit when traded
3. Sell a token that already has approval
* Should NOT ask for allowance signature
* Should place order
* Should NOT contain prehook in appData
